poollet/machinepoollet/mem/mem.go

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package mem
+
+import (
+	"context"
+	"errors"
+
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var ErrNoMatchingMachineEvents = errors.New("no matching machine events")
+
+type MachineEventMapper interface {
+	manager.Runnable
+	GetMachineEventFor(ctx context.Context, machineID string) ([]*iri.Event, error)
```

Comment:
Do we really need this function? atm. it's only used in the tests

Reply:
removed GetMachineEventFor() method